daemon/cmd/main.go

```diff
@@ -0,0 +1,222 @@
+)
+
+const (
+	serviceName = "OCDaemon-PHP"
+
+	defaultOCAgentAddr      = "localhost:55678"
+	defaultUnixSocketPath   = "/tmp/ocdaemon.sock"
+	defaultLogLevel         = logError
+	defaultMsgBufSize       = 1000
+	defaultZPagesAddr       = ":8888"
+	defaultZPagesPathPrefix = "/debug"
+)
+
+const (
+	logNone = iota
+	logError
+	logWarn
+	logInfo
+	logDebug
+)
+
+var (
+	buildVersion string
+	buildDate    string
+)
+
+func main() {
+	var (
+		fs                   = flag.NewFlagSet("", flag.ExitOnError)
+		flagOCAgentAddr      = fs.String("ocagent.addr", defaultOCAgentAddr, "Address of the OpenCensus Agent")
+		flagServiceName      = fs.String("php.servicename", os.Getenv("HOSTNAME"), "Name of our PHP service")
+		flagLogLevel         = fs.Int("log.level", defaultLogLevel, "Logging level to use")
+		flagMsgBufSize       = fs.Int("msg.bufsize", defaultMsgBufSize, "Size of buffered message channel")
+		flagZPagesAddr       = fs.String("zpages.addr", defaultZPagesAddr, "zPages bind address")
+		flagZPagesPathPrefix = fs.String("zpages.path", defaultZPagesPathPrefix, "zPages path prefix")
+		flagUnixSocketPath   = fs.String("socket.path", defaultUnixSocketPath, "Unix socket path to listen on")
+		flagVersion          = fs.Bool("version", false, "Show version information")
+
+		processor      daemon.Processor
+		closeRequested bool
+	)
+
+	// parse our command line override flags
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		_, _ = fmt.Fprintf(os.Stderr, "%v", err)
+		os.Exit(1)
+	}
+
+	if *flagVersion {
+		fmt.Printf(
+			"Version information for: %s\nBuild     : %s\nBuild date: %s\n\n",
+			serviceName,
+			buildVersion,
+			buildDate,
+		)
+		os.Exit(0)
+	}
+
+	var logger log.Logger
+	{
+		logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+		switch *flagLogLevel {
+		case logNone:
+			logger = log.NewNopLogger()
+		case logError:
+			logger = level.NewFilter(logger, level.AllowError())
+		case logWarn:
+			logger = level.NewFilter(logger, level.AllowWarn())
+		case logInfo:
+			logger = level.NewFilter(logger, level.AllowInfo())
+		case logDebug:
+			logger = level.NewFilter(logger, level.AllowDebug())
+		default:
+			logger = level.NewFilter(logger, level.AllowError())
+		}
+		logger = log.With(logger,
+			"svc", serviceName,
+			"ts", log.DefaultTimestampUTC,
+			"clr", log.DefaultCaller,
+		)
+
+	}
+
+	_ = level.Info(logger).Log("msg", "service started", "agent", *flagOCAgentAddr)
+	defer func() {
+		_ = level.Info(logger).Log("msg", "service ended")
+	}()
+
+	var g run.Group
+	if *flagZPagesAddr != "" {
+		// set-up zPages
+		var (
+			l   net.Listener
+			err error
+			mux = http.NewServeMux()
+		)
+		zpages.Handle(mux, *flagZPagesPathPrefix)
+		l, err = net.Listen("tcp", *flagZPagesAddr)
+
+		if err != nil {
+			_ = level.Error(logger).Log("msg", "failed to bind to requested zPages address", "err", err)
+			os.Exit(1)
+		}
+		g.Add(func() error {
+			return http.Serve(l, mux)
+		}, func(error) {
+			_ = l.Close()
+		})
+	}
+	{
```

Comment:
Scoping manually by using braces is more useful when you have variables that might clash. However, currently I don't see any. Could you please remove these? I believe the newlines and the section comments that you've already provided suffice.

Reply:
It's a coding style thing (often used in Go kit projects). I like using them (especially in main functions) as they not only guard from potential variable clashes but also work as a grouping of code  (they can even be collapsed in most IDE's). They clearly show the various sections within the main func while preserving a top to bottom directional flow. I've looked at what happens if I remove them and I find the code becoming less readable. Another alternative would be to move these sections into their own functions. In this case that would work well too.

Reply:
I ended up placing the various services into their own init functions removing all scoping braces.